File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.cpp
class Solution {
public:
    //Time complexity: O(n)
    //Space complexity:O(1)
    long long minimumSteps(string s) {
        long long ans{0}, lastZeroIndex{-1};
        
        for(int i = 0; i < s.size(); i++){
            if(s[i] == '0'){
                int steps = i-lastZeroIndex-1;
                lastZeroIndex++;
                ans += steps;
            }
        }
        
        return ans;
    }
};

File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.py
class Solution:
   #Time complexity: O(N)
   #Space complexity: O(1)
    def minimumSteps(self, s: str) -> int:
        l, r = 0, len(s) - 1
        ans = 0
        
        while l < r:
            while l < r and s[l] == '0':
                l += 1
            while l < r and s[r] == '1':
                r -= 1
            ans += r - l
            l += 1
            r -= 1
        return ans